### ISSUE TITLE
I'm in, and I'm doing horrible things to fanfiction.

### ISSUE THREAD

**Issue by ianrenton** [dariusk/NaNoGenMo#60]
My plan involves Fanfiction.net, and it is going to be _awful_.

**Comment by ianrenton** [comment]
The script is coming along nicely. It's like one of those viagra spam mails that try not to look like spam - only it's about Doctor Who.

Here's the repository: https://github.com/ianrenton/NaNoGenMo

And an example of what it currently generates: http://pastebin.com/9q62dtwr

**Comment by lilinx** [comment]
this could have fooled me, I didn't know there were Viagra spam generated in a similar way

**Comment by ianrenton** [comment]
I think it's a new technique that's still relatively rare among spammers - I've only received a few messages like it.

The technique is called "Bayesian Poisoning" (according to Wikipedia, anyway) and tries to both evade and degrade the performance of Bayesian spam filters.

http://en.wikipedia.org/wiki/Bayesian_poisoning

**Comment by dariusk** [comment]
Wow, very very cool.

**Comment by jseakle** [comment]
> I think it's a new technique that's still relatively rare among spammers

I don't think this is the case. I was under the impression that _most_ spam these days uses Bayesian Poisoning; at the very least I've personally been receiving it in bulk for at least five years (I think more).

**Comment by aandnota** [comment]
This is great!

**Comment by ianrenton** [comment]
@jseakle Fair enough - I only assumed it was rare because I rarely receive that kind of spam, and the Wikipedia page was pretty difficult to find since I didn't know the name of it!

**Comment by ianrenton** [comment]
Since the last example I've added HTML generation using an ERB template, and the story can now be split into a number of chapters.

I've put up a new example output on GitHub Pages here: http://ianrenton.github.io/NaNoGenMo/example.html

Advance warning: this was generated from My Little Pony / Sonic the Hedgehog cross-over fanfiction, so it's... special. Not in a good way.

**Comment by ianrenton** [comment]
Well, November's gone so I guess I should call time on this script! Thanks to everyone who's offered tips and praise for my junk code, and every poor fool who has incorporated my code into their own NaNoGenMo script!
